core/libs/shared/src/core_ops.rs

```diff
@@ -670,11 +696,32 @@ where
                             let remote_deleted = remote.calculate_deleted(&id)?;
                             let local = remote.stage(local_changes);
                             result = local.stage(merge_changes);
-                            (remote_change, remote_name, remote_deleted, local_change)
+                            let user_access_keys = merge_user_access(
+                                None,
+                                remote_change.user_access_keys(),
+                                local_change.user_access_keys(),
+                            );
+                            (
+                                remote_change,
+                                remote_name,
+                                remote_deleted,
+                                local_change,
+                                user_access_keys,
+                            )
                         };
 
                         let key = result.decrypt_key(&id, account)?;
-                        let parent_key = result.decrypt_key(remote_change.parent(), account)?;
+                        let name = {
+                            // we may not have the parent of a direct share
+                            // in that case changes are unauthorized anyway
```

Comment:
before this change, any concurrent edit on a directly shared file would make sync persistently fail